enhancements/ingress/gateway-api-crd-life-cycle-management.md

```diff
@@ -0,0 +1,472 @@
+these CRDs.
+
+## Motivation
+
+Gateway API is an evolving API, and its CRDs change over time.  For core
+OpenShift, we must ensure that we have a version of the CRDs that is compatible
+with our product, and we must handle upgrading these CRDs for our own needs.  We
+must also avoid unnecessarily causing friction for Red Hat layered products,
+partner products, and other third-party implementations.
+
+### User Stories
+
+#### Using OpenShift's Gateway API implementation on a new OpenShift cluster
+
+As a cluster-admin, I want to install a new OpenShift 4.19 cluster, and then use
+OpenShift's Gateway API implementation to configure ingress to workload on this
+cluster.
+
+_TBD_
+
+#### Using OpenShift's Gateway API implementation on an upgraded cluster
+
+As a cluster-admin, I want to upgrade my cluster from OpenShift 4.18 (which
+*doesn't* manage the Gateway API CRDs' life-cycle) to OpenShift 4.19 (which
+*does*).
+
+_TBD_
+
+#### Using a third-party Gateway API implementation
+
+As a cluster-admin, I want to install a third-party Gateway API implementation
+on my OpenShift 4.19 cluster, and I want to use the third-party implementation
+without interference from OpenShift.
+
+_TBD_
+
+#### Future OpenShift upgrades
+
+As a cluster-admin, I want to upgrade my cluster from OpenShift 4.19 (which
+manages the Gateway API CRDs' life-cycle) to OpenShift 4.20 (which includes a
+newer version of these CRDs).
+
+_TBD_
+
+### Goals
+
+- Ensure Gateway API CRDs are installed on new and upgraded OpenShift 4.19 clusters.
+- Ensure the installed Gateway API CRDs are compatible with OpenShift's needs.
+- Detect and warn if an incompatible version of the CRDs is installed.
+- Describe the issue of dead fields and potential risks thereof.
+
+### Non-Goals
+
+- Automatically replace incompatible CRDs that some other agent installed.
+- Provide a pre-upgrade check in OpenShift 4.18 for incompatible CRDs.
+- Provide an explicit override for the cluster-admin to take CRD ownership.
+- Implement an admission webhook to block updates to dead fields.
+- Solve CRD life-cycle management in OLM.
+- Solve OLM subscription management.
+- Solve CRD life-cycle management for OSSM or Istio resources.
+
+## Proposal
+
+### Life-cycle management
+
+OpenShift's Ingress Operator monitors for the presence of the Gateway API CRDs.
+
+- If they are absent, it installs the appropriate version.
+- If they are present at the expected version, the operator does nothing.
+- If they are present at an unexpected version, the operator signals a degraded state.
+
+Note that the appropriate version for the CRDs depends on the version of
+OpenShift Service Mesh (OSSM) that the Ingress Operator has pinned in a given
+OpenShift release.  Specifically, the CRDs SHOULD be the version corresponding
+to the version of Istio in that OSSM version; they MUST be of some version that
+is compatible and that we have tested with that version of Istio.
```

Comment:
```suggestion
OpenShift release.  Specifically, the CRDs SHOULD be the version corresponding
to the version of Istio in that OSSM version; they MUST be of some version that
is compatible and that we have tested with that version of Istio (TODO: expand upon this)
```

Reply:
Done in https://github.com/openshift/enhancements/compare/99aae59bb1ac50f1ef02e07ff3dcc3385216c451..e34441f0d0685989e6417db70dcf049c68f9fc55.  Is that sufficient?  If we need more detail, it might warrant a separate section for how we validate a particular CRD version.